### DISCUSSION TITLE
Traffic simulation from Python

### DISCUSSION THREAD

**Discussion by rmonstein** [TUDelft-CNS-ATM/bluesky#319] [Ideas]
Hi you good people at bluesky,

we are considering (ab-)using bluesky to generate trajectories but we would like to use it more like a Python package instead of a standalone application. To that end, I developed a small prototype to see if I can get it working (it seems like I can).

We would like to use bluesky to generate trajectories that will then be used in Monte Carlo simulations. In order for it to be feasible, we would need to generate a couple of million trajectories in a reasonable amount of time. Since we don't need a GUI or any other features besides the generation and simulation of traffic, we would like to bypass all that and call the internals of bluesky directly.

In case anybody is interested, I can post the Python script, just let me know. I would appreciate a feedback on what we're trying from somebody with a better knowledge of bluesky.

Now to my question:
I found a few minor inconsistencies and bugs in the code. Is there an interest that I share my fixes and maybe an example of my use case with the community (i.e., should I make a PR)?

**Comment by jooste** [comment]
Hi,

Thanks for the suggestion! It is indeed possible to just import bluesky as a module (as you obviously found out 😉). There are several forks out there that do it as well, mostly to use it with a reinforcement learning environment. It'd be nice though to either have a little tutorial on the wiki, or a gist to link to.

And as for bugs and inconsistencies; we have no illusions that our code is bug-free 😁, and good fixes are very welcome as PRs!

**Reply by rmonstein** [reply]
I tried to make a PR for my changes, but I don't have the privileges to create a new branch. How should I submit my changes?

**Reply by jooste** [reply]
You should be able to create a PR with the master version of BlueSky from your own fork, have you tried that?

**Reply by rmonstein** [reply]
I see, I somehow missed the "compare across forks" option. Thanks

**Comment by rmonstein** [comment]
Thanks for merging the changes. Should I write a short wiki entry to point the interested reader to the Jupyter Notebook file in the _Docs_ folder?

**Comment by jooste** [comment]
That would be great! I just added an entry in the wiki table of contents: Importing BlueSky as a python package. Could you make a page that this entry then links to? The prettiest way would be to render the notebook within a wiki page, but I couldn't find a way to do this, so I don't know if it is possible!

**Reply by rmonstein** [reply]
I just added a short page with basically only the link to the jupyter notebook. It would be nice to embed it in the wiki page directly, but I couldn't find a way to do that. If anybody has an idea, feel free to edit.
